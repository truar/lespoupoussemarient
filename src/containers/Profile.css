.Profile {
    background: url('../images/login.jpg') no-repeat center fixed;
    background-size: cover; /* version standardis√©e */
    height: 100vh;
}

.Profile .WrapperContent {
    justify-content: center;
    flex-grow: 2;
}

.Profile form {
    display: flex;
    flex-direction: column;
    flex-grow: 0.1;
    justify-content: center;
    width: 90%;
    align-items: center;
}

.Profile .form-group {
    width: 100%;
    justify-content: center;
}

.Profile .invalid-feedback {
    font-size: 1.3rem;
    text-align: center;
    font-weight: bold;
    color: white;
}

@media(min-width: 768px) {
    .Profile form {
        width: 25%;
    }
}