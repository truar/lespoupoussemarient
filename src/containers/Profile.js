import React, { Component } from 'react';
import {
    Form, FormGroup, FormFeedback, Button, Input
  } from 'reactstrap';
import Section from '../components/Section';
import WrapperContent from '../components/WrapperContent';
import './Profile.css';

class Profile extends Component {
    calculateRemainingDays() {
        var dateWedding = new Date(2019, 7, 31);
        var today = Date.now();
        var numOfDays = Math.round((dateWedding-today)/(1000*60*60*24));
        var dateDisplay = "J-" + numOfDays;
        if(numOfDays === 0) {
            dateDisplay = "JOUR-J";
        } else if(numOfDays < 0) {
            dateDisplay = "Mariage terminÃ©";
        }
        return dateDisplay;
    }

    render() {
        const {isLogged, ...attributes} = this.props;
        const toRender = (isLogged) ? this.renderLogged() : this.renderAuthenticationForm();
        return (
            <Section id="Profile" className="Profile" {...attributes}>
                {toRender}
            </Section>
        );
    }

    renderAuthenticationForm() {
        const {error, guestCode, onChange, onSubmit} = this.props;
        return (
            <WrapperContent title="Les poupous se marient !">
                <Form onSubmit={onSubmit} id="login-form">
                    <FormGroup>
                        <Input invalid={error} onChange={onChange} type="text" name="guestCode" id="guestCode" placeholder="Entrez votre code..." value={guestCode}/>
                        <FormFeedback>Code incorrect</FormFeedback>
                    </FormGroup>
                    <Button>Se connecter</Button>
                </Form>
            </WrapperContent>
        );
    }

    renderLogged() {
        const dateDisplay = this.calculateRemainingDays();
        const profile = this.props.profile;
        let displayText = <p>Nous avons le plaisir de vous convier au <b>Vin d'honneur</b> !</p>;
        if(profile === 2) {
            displayText = <p>Nous avons le plaisir de vous convier au <b>Vin d'honneur et Diner</b> !</p>;
        } else if(profile === 3) {
            displayText = <p>Nous avons le plaisir de vous convier au <b>Vin d'honneur, Diner et Brunch</b> !</p>;
        }
        return (
            <WrapperContent title="Que l'aventure commence !">
                <p>{dateDisplay}</p>
                {displayText}
            </WrapperContent>
        );
    }
}

export default Profile;