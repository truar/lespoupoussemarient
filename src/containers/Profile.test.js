import React from 'react';
import Profile from './Profile';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<Profile />);
  expect(component).toMatchSnapshot();
});

