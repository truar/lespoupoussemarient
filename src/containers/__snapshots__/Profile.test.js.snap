// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WrapperContent
        title="Les poupous se marient !"
      >
        <Form
          id="login-form"
          tag="form"
        >
          <FormGroup
            tag="div"
          >
            <Input
              id="guestCode"
              name="guestCode"
              placeholder="Entrez votre code..."
              type="text"
            />
            <FormFeedback
              tag="div"
            >
              Code incorrect
            </FormFeedback>
          </FormGroup>
          <Button
            color="secondary"
            tag="button"
          >
            Se connecter
          </Button>
        </Form>
      </WrapperContent>,
      "className": "Profile",
      "id": "Profile",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Form
          id="login-form"
          tag="form"
        >
          <FormGroup
            tag="div"
          >
            <Input
              id="guestCode"
              name="guestCode"
              placeholder="Entrez votre code..."
              type="text"
            />
            <FormFeedback
              tag="div"
            >
              Code incorrect
            </FormFeedback>
          </FormGroup>
          <Button
            color="secondary"
            tag="button"
          >
            Se connecter
          </Button>
        </Form>,
        "title": "Les poupous se marient !",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FormGroup
              tag="div"
            >
              <Input
                id="guestCode"
                name="guestCode"
                placeholder="Entrez votre code..."
                type="text"
              />
              <FormFeedback
                tag="div"
              >
                Code incorrect
              </FormFeedback>
            </FormGroup>,
            <Button
              color="secondary"
              tag="button"
            >
              Se connecter
            </Button>,
          ],
          "id": "login-form",
          "onSubmit": undefined,
          "tag": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Input
                  id="guestCode"
                  name="guestCode"
                  placeholder="Entrez votre code..."
                  type="text"
                />,
                <FormFeedback
                  tag="div"
                >
                  Code incorrect
                </FormFeedback>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "guestCode",
                  "invalid": undefined,
                  "name": "guestCode",
                  "onChange": undefined,
                  "placeholder": "Entrez votre code...",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Code incorrect",
                  "tag": "div",
                  "valid": undefined,
                },
                "ref": null,
                "rendered": "Code incorrect",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Se connecter",
              "color": "secondary",
              "tag": "button",
            },
            "ref": null,
            "rendered": "Se connecter",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WrapperContent
          title="Les poupous se marient !"
        >
          <Form
            id="login-form"
            tag="form"
          >
            <FormGroup
              tag="div"
            >
              <Input
                id="guestCode"
                name="guestCode"
                placeholder="Entrez votre code..."
                type="text"
              />
              <FormFeedback
                tag="div"
              >
                Code incorrect
              </FormFeedback>
            </FormGroup>
            <Button
              color="secondary"
              tag="button"
            >
              Se connecter
            </Button>
          </Form>
        </WrapperContent>,
        "className": "Profile",
        "id": "Profile",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Form
            id="login-form"
            tag="form"
          >
            <FormGroup
              tag="div"
            >
              <Input
                id="guestCode"
                name="guestCode"
                placeholder="Entrez votre code..."
                type="text"
              />
              <FormFeedback
                tag="div"
              >
                Code incorrect
              </FormFeedback>
            </FormGroup>
            <Button
              color="secondary"
              tag="button"
            >
              Se connecter
            </Button>
          </Form>,
          "title": "Les poupous se marient !",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <FormGroup
                tag="div"
              >
                <Input
                  id="guestCode"
                  name="guestCode"
                  placeholder="Entrez votre code..."
                  type="text"
                />
                <FormFeedback
                  tag="div"
                >
                  Code incorrect
                </FormFeedback>
              </FormGroup>,
              <Button
                color="secondary"
                tag="button"
              >
                Se connecter
              </Button>,
            ],
            "id": "login-form",
            "onSubmit": undefined,
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Input
                    id="guestCode"
                    name="guestCode"
                    placeholder="Entrez votre code..."
                    type="text"
                  />,
                  <FormFeedback
                    tag="div"
                  >
                    Code incorrect
                  </FormFeedback>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "guestCode",
                    "invalid": undefined,
                    "name": "guestCode",
                    "onChange": undefined,
                    "placeholder": "Entrez votre code...",
                    "type": "text",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Code incorrect",
                    "tag": "div",
                    "valid": undefined,
                  },
                  "ref": null,
                  "rendered": "Code incorrect",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Se connecter",
                "color": "secondary",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Se connecter",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
